vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -165,20 +165,23 @@
         }
 
         grid.selectedItems = grid.selectedItems.concat(items);
-        items.forEach(item => {
-          if (item) {
-            selectedKeys[item.key] = item;
-            if (userOriginated) {
-              item.selected = true;
-              grid.$server.select(item.key);
+        const areAllElementOfItemsNotNull = items.every(function(i) { return i !== null; });
+        if (areAllElementOfItemsNotNull) {
+          items.forEach(item => {
+            if (item) {
+              selectedKeys[item.key] = item;
+              if (userOriginated) {
+                item.selected = true;
+                grid.$server.select(item.key);
+              }
             }
-          }
-          const isSelectedItemDifferentOrNull = !grid.activeItem || !item || item.key != grid.activeItem.key;
-          if (!userOriginated && selectionMode === 'SINGLE' && isSelectedItemDifferentOrNull) {
-            grid.activeItem = item;
-            grid.$connector.activeItem = item;
-          }
-        });
+            const isSelectedItemDifferentOrNull = !grid.activeItem || !item || item.key != grid.activeItem.key;
+            if (!userOriginated && selectionMode === 'SINGLE' && isSelectedItemDifferentOrNull) {
+              grid.activeItem = item;
```

Comment:
Now that this isn't run for `null` value, `grid.activeItem` has an invalid value after deselection. It should be `null` but it still has an object reference. The fix should probably be done in `__activeItemChanged`

Reply:
have a new commit to fix the bug in __activeItemChanged.
Explain the fix: after investigating the bug, the bug comes from when setItems after select(items), the key of select items still there, and original code from master, switch the oldVal and newVal in if (!grid.$connector.deselectAllowed && oldVal) { ..... } , the condition is triggered for setItems, in observer __activeItemChanged. With this switch, it will go else if (!selectedKeys[newVal.key]) { grid.$connector.doSelection([newVal], true); } => key is unknown => bug.
My idea is if the case is raised , I call doDeselection with item is null and it will do nothing in doDeselection method.

Reply:
Why would you want to call `doDeselection` with `[null]`if it does nothing??? The result is the same as removing the line altogether.

Also, now that you removed `grid.activeItem = oldVal;`, active item gets set to `null` while there still is a selected item (`activeItem` and `selectedItems` are no longer in sync).

Reply:
new fix. PTAL it.